### PR TITLE
Add `F.roi_average_pooling_2d`

### PR DESCRIPTION
**Pull request by knorth55**
EDIT:
- add `roi_average_pooling_2d`
~~- rename `roi_pooling_2d` -> `roi_max_pooling_2d`~~
~~- add `roi_pooling_2d` with `operation` arg for interface~~

### CODE REVIEW

File: chainer/functions/pooling/roi_average_pooling_2d.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+            }
+            ''', 'roi_average_pooling_2d_bwd'
+        )(gy[0], bottom_rois, self.spatial_scale,
+          channels, height, width, self.outh, self.outw,
+          bottom_diff, size=gy[0].size)
+
+        return bottom_diff, None
+
+
+def roi_average_pooling_2d(x, rois, outh, outw, spatial_scale):
```
Comment (beam2d) [inline chainer/functions/pooling/roi_average_pooling_2d.py]: Can we use a pair of ints (a tuple of two ints) for `outh, outw` for consistency with other functions? For example, `unpooling_2d` and `upsampling_2d` have `outsize` parameter which accepts a pair of ints that specifies the output image size (in these functions, `outsize` is optional, so strictly speaking, the requirements of the API design is a bit different, though).

File: chainer/functions/pooling/roi_max_pooling_2d.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            bottom_diff = gradient;
+            ''', 'roi_max_pooling_2d_bwd'
+        )(gy[0], self.argmax_data, bottom_rois.shape[0], self.spatial_scale,
+          channels, height, width, self.outh, self.outw,
+          bottom_rois, bottom_diff)
+
+        return bottom_diff, None
+
+
+def roi_max_pooling_2d(x, rois, outh, outw, spatial_scale):
```
Comment (beam2d) [inline chainer/functions/pooling/roi_max_pooling_2d.py]: ditto

**Comment by knorth55** [comment]
I updated to new interface.
roi_average_pooling_2d

File: chainer/functions/__init__.py [hunk 3]
```diff
@@ -203,6 +203,7 @@
 # https://github.com/chainer/chainer/pull/5198
 from chainer.functions.pooling.roi_align_2d import roi_align_2d as roi_average_align_2d  # NOQA
 from chainer.functions.pooling.roi_pooling_2d import roi_pooling_2d  # NOQA
+from chainer.functions.pooling.roi_average_pooling_2d import roi_average_pooling_2d  # NOQA
```
Comment (Hakuyume) [inline chainer/functions/__init__.py]: alphabetic order

Comment (knorth55) [reply chainer/functions/__init__.py]: updated

**Comment by Hakuyume** [comment]
jenkins, test this please.

**Comment by knorth55** [comment]
@Hakuyume can you review this PR?

**Comment by Hakuyume** [comment]
jenkins, test this please

**Comment by knorth55** [comment]
@Hakuyume any updates?

**Comment by knorth55** [comment]
Any updates?

**Comment by kmaehashi** [comment]
I confirmed that comment from @beam2d has been resolved.

**Pull request merged**